chainercv/links/detection_link.py

```diff
@@ -0,0 +1,39 @@
+import chainer
+
+
+class DetectionLink(chainer.Link):
```

Comment:
I am not a big fan of double inheritance in general because its `__init__` is confusing.
Also, it may give an impression that ChainerCV builds a lot of abstraction on top of Chainer for using a complex OO design.

On the other hand, I can see benefit of using multiple inheritance, too.

I am probably not convinced because there isn't any example that seems to benefit from extending from `ChainList` or `Link` directly with `DetectionList`.